### PR TITLE
xds: set dial target in request resource_names

### PR DESCRIPTION
**Pull request by menghanl**
And stop sending metadata `TRAFFICDIRECTOR_GRPC_HOSTNAME`.
Similarly, for LRS, cluster_name is set to the dial target.

---

The expected resource_names can be in a different format from dial target. (test_service.test_namespace.traffic_director.com vs test_namespace:test_service).  The mapping will be done by LDS/RDS/VHDS + CDS.

The workaround was to leave resource_names empty, and send dial target as metadata `TRAFFICDIRECTOR_GRPC_HOSTNAME`.
This change sets dial target directly in resource_names, and removes metadata. The server don't need to read the metadata for service name, but needs to support two formats of resrouce_names (test_service.test_namespace.traffic_director.com and test_namespace:test_service).

When all the missing pieces are done, resource_names will be set as the cluster names from CDS response.
For LRS, cluster_name will be set to the same name.

### CODE REVIEW

**Comment by menghanl** [comment]
PR description updated.

File: xds/internal/balancer/lrs/lrs.go [hunk 1]
```diff
@@ -152,15 +151,7 @@ type lrsStore struct {
...
 	return &lrsStore{
-		node: &basepb.Node{
-			Metadata: &structpb.Struct{
-				Fields: map[string]*structpb.Value{
-					internal.GrpcHostname: {
-						Kind: &structpb.Value_StringValue{StringValue: serviceName},
-					},
-				},
-			},
-		},
```
Comment (easwars) [inline xds/internal/balancer/lrs/lrs.go]: Will LRS be eventually moved to the xdsClient and will a nodeProto (read from the bootstrap file) be added to the request there? If not, is OK to not have a nodeProto at all in the LRS request?

Comment (menghanl) [reply xds/internal/balancer/lrs/lrs.go]: Yes, LRS will be moved to the same xdsClient.

Comment (easwars) [reply xds/internal/balancer/lrs/lrs.go]: Should we add a TODO to say that we should add the NodeProto here once we move it to the xdsClient.

Comment (menghanl) [reply xds/internal/balancer/lrs/lrs.go]: Done

**Pull request merged**